jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebAppClassLoader.java

```diff
@@ -223,23 +219,30 @@ public Context getContext()
     }
 
     /**
-     * @param resources The resources to add to the classpath
-     * @throws IOException if unable to add classpath from resource
+     * @param resource The resources to add to the classpath
      */
-    public void addClassPath(Resource resources)
-        throws IOException
+    public void addClassPath(Resource resource)
     {
-        for (Resource resource: resources)
+        ResourceCollection.stream(resource).forEach(r ->
```

Comment:
Since any `Resource` might be a collection, if we need URLs, URIs or Paths from a resource, then we should use this stream abstraction to get back the one or more values.  The stream can then be used to pick the first, collect into a list or do forEach handling like I am below: